src/convertToObject.js

```diff
@@ -1,12 +1,18 @@
-'use strict';
-
-/**
- * @param {string} sourceString
- *
- * @return {object}
- */
 function convertToObject(sourceString) {
-  // write your code here
+  const styleEntries = sourceString
+    .split(';')
+    .map((item) => item.trim())
+    .filter((item) => item)
+    .map((item) => item.split(':').map((subItem) => subItem.trim()));
+
+  const styleObject = styleEntries.reduce((obj, [key, value]) => {
+    return {
+      ...obj,
+      [key]: value,
+    };
+  }, {});
```

Comment:
To improve efficiency and avoid unnecessary creation of new objects, it is better to change this part like this:

```suggestion
  const styleObject = styleEntries.reduce((obj, [key, value]) => {
    if (key && value) {
      obj[key] = value;
    }
    return obj;
 }, {});
```

Explanation: 
The `reduce` method is used to accumulate the styles object.
The ``if (key && value)`` test ensures that only those pairs that have both a ``key`` and a `value` are added to the object. This prevents possible errors when processing incorrect declarations.